app/controllers/proposal/approved_proposals_controller.rb

```diff
@@ -0,0 +1,47 @@
+module Proposal
+  class ApprovedProposalsController < ManageMetadataController
+    # There are exceptions in this code for proposal_mode checks for development
+    # and test to prevent the need to run two servers while using those
+    # environments.
+
+    skip_before_action :ensure_user_is_logged_in
+    skip_before_action :refresh_urs_if_needed, :refresh_launchpad_if_needed, :provider_set?, :proposal_approver_permissions
+
+    def approved_proposals
+      token_and_client_id = request.headers.fetch('Echo-Token', ':').split(':')
+      passed_token = token_and_client_id[0]
+      passed_client_id = token_and_client_id[1]
+
+      # Navigate a browser elsewhere
+      redirect_to root_path and return if passed_token.blank?
+
+      token_response = cmr_client.validate_token(passed_token, passed_client_id)
+
+      if token_response.success?
+        authenticated_user = User.new(urs_uid: token_response.body['uid'])
+        requester_has_approver_permissions = is_non_nasa_draft_approver?(user: authenticated_user, token: passed_token)
+      else
+        requester_has_approver_permissions = false
+      end
+
```

Comment:
rubocopy didn't complain here?   
requester_has_approver_permissons = if ... else ...

Reply:
No, the first part of the if has two statements.  I've reworked it to address that though.

Reply:
I think you can do something like this, correct if me I'm wrong:

```
    requester_has_approver_permissions = if token_response.success?
                                           authenticated_user = User.new(urs_uid: token_response.body['uid'])
                                           is_non_nasa_draft_approver?(user: authenticated_user, token: passed_token)
                                         else
                                           false
                                         end
```

Reply:
I think that would work too, but I changed the later statement that was relying on the authenticated user, so this has been changed slightly differently.
Another way of stating the first sentence of my other comment is: I think rubocop only balks in the most trivial cases.  I didn't mean to imply it wouldn't work.